### PR TITLE
Update backoffice and marketplace contract to add hackathons endpoints

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml [hunk 1]
```diff
@@ -1580,6 +1580,111 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HackathonsDetailsResponse'
+        "401":
+          $ref: '#/components/responses/UnauthorizedError'
+        "500":
+          $ref: '#/components/responses/InternalServerError'
+
+  /bo/v1/hackathons/id/{hackathonId}:
```
Comment (ofux) [inline backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml:1656]:
```suggestion
  /bo/v1/hackathons/{hackathonId}:
```

File: backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml [hunk 2]
```diff
@@ -2080,6 +2185,30 @@ components:
...
+
+    HackathonSlug:
+      type: string
+
+    HackathonStatus:
+      type: string
+      enum:
+        - DRAFT
+        - PUBLISHED
+        - CANCELED
```
Comment (ofux) [inline backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml:2200]: Not sure we need this one?

File: backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml [hunk 3]
```diff
@@ -1580,6 +1580,111 @@ paths:
...
+              $ref: '#/components/schemas/UpdateHackathonRequest'
+      responses:
+        "204":
+          description: 'The hackathon has been updated'
+        "401":
+          $ref: '#/components/responses/UnauthorizedError'
+        "500":
+          $ref: '#/components/responses/InternalServerError'
+
+
```
Comment (ofux) [inline backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml:1687]: Shouldn't we add a DELETE as well?

**Review by ofux — CHANGES_REQUESTED** [review]

File: backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml (hunk 2 above)
Comment (alexbeno) [reply backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml:2200]: Need for display and update ?

File: backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml (hunk 3 above)
Comment (alexbeno) [reply backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml:1687]: We don't know if we want to delete or just cancel/archive it, has a CMS will do. 

We can have a case that the hackathon could be deleted but the team want to keep it as archived ?

File: backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml (hunk 2 above)
Comment (ofux) [reply backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml:2200]: I was speaking only about the `CANCELED` value.
I'm not sure we want to deal with soft-deletes for now. Maybe we can handle it later if needed. I would rather implement a simple hard-delete for now. WDYT?

Comment (alexbeno) [reply backoffice-api-contract/src/main/resources/backoffice-api-contract.yaml:2200]: Ok let's do it

**Review by ofux — APPROVED** [review]

**Review by PierreOucif — APPROVED** [review]

**Pull request merged**